osutil/kcmdline.go

```diff
@@ -345,3 +353,81 @@ func KernelCommandLine() (string, error) {
 	}
 	return strings.TrimSpace(string(buf)), nil
 }
+
+type ValuePattern interface {
+	Match(value string) bool
+}
+
+type ValuePatternAny struct {
+}
+
+func (any ValuePatternAny) Match(value string) bool {
+	return true
+}
+
+type ValuePatternConstant struct {
+	constantValue string
+}
+
+func NewConstantValuePattern(constant string) ValuePatternConstant {
+	return ValuePatternConstant{constant}
+}
+
+func (constant ValuePatternConstant) Match(value string) bool {
+	return constant.constantValue == value
+}
+
+type KernelArgumentPattern struct {
+	Param string
+	Value ValuePattern
+}
+
+func (kap *KernelArgumentPattern) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var arg string
+	if err := unmarshal(&arg); err != nil {
+		return errors.New("cannot unmarshal kernel argument")
+	}
+
+	parsed := ParseKernelCommandline(arg)
+	if len(parsed) != 1 {
+		return fmt.Errorf("%q is not a unique kernel argument", arg)
+	}
+	// To make parsing future proof in case we support full
+	// globbing in the future, do not allow unquoted globbing
+	// characters, except the currently only supported case ('*').
+	if !parsed[0].Quoted && parsed[0].Value != "*" &&
+		strings.ContainsAny(parsed[0].Value, `*?[]\{}`) {
+		return fmt.Errorf("%q contains globbing characters and is not quoted",
+			parsed[0].Value)
+	}
+	kap.Param = parsed[0].Param
+	if parsed[0].Quoted || parsed[0].Value != "*" {
+		kap.Value = ValuePatternConstant{parsed[0].Value}
```

Comment:
I find a bit strange to not use `NewConstantValuePattern` while using it in the tests. I understand you want to use this method specifically to encapsulate building structs in the tests, but then the name should reflect it more clearly that it should be used only in unit tests.

Reply:
I have refactored the code. Now I introduced higher level functions `NewConstantKernelArgumentPattern` and `NewAnyKernelArgumentPattern`. Since those have to be used in tests in other packages, I cannot hide them in `export_test.go`. Still have I have commented on them that they are mainly used for tests. But since they are higher level, it would be fine if they were used other than tests. I have refactored lower level types and fields to make them private.